dipy/viz/utils.py

```diff
@@ -0,0 +1,96 @@
+
+from __future__ import division, print_function, absolute_import
+
+import numpy as np
+from scipy.ndimage import map_coordinates
+
+# Conditional import machinery for vtk
+from dipy.utils.optpkg import optional_package
+
+# import vtk
+# Allow import, but disable doctests if we don't have vtk
+vtk, have_vtk, setup_module = optional_package('vtk')
+ns, have_numpy_support, _ = optional_package('vtk.util.numpy_support')
+
+
+def numpy_to_vtk_points(points):
+    """ numpy points array to a vtk points array
+    """
+    vtk_points = vtk.vtkPoints()
+    vtk_points.SetData(ns.numpy_to_vtk(np.asarray(points), deep=True))
+    return vtk_points
+
+
+def numpy_to_vtk_colors(colors):
+    """ numpy color array to a vtk color array
+
+    if colors are not already in UNSIGNED_CHAR
+        you may need to multiply by 255.
+
+    Example
+    ----------
+    >>>  vtk_colors = numpy_to_vtk_colors(255 * float_array)
+    """
+    vtk_colors = ns.numpy_to_vtk(np.asarray(colors), deep=True,
+                                 array_type=vtk.VTK_UNSIGNED_CHAR)
+    return vtk_colors
+
+
+def set_input(vtk_object, inp):
+    """ Generic input function which takes into account VTK 5 or 6
+
+    Parameters
+    ----------
+    vtk_object: vtk object
+    inp: vtkPolyData or vtkImageData or vtkAlgorithmOutput
+
+    Returns
+    -------
+    vtk_object
+
+    Example
+    ----------
+    >>> poly_mapper = set_input(vtk.vtkPolyDataMapper(), poly_data)
+    """
+    if isinstance(inp, vtk.vtkPolyData) \
+       or isinstance(inp, vtk.vtkImageData):
+        if vtk.VTK_MAJOR_VERSION <= 5:
+            vtk_object.SetInput(inp)
+        else:
+            vtk_object.SetInputData(inp)
+    elif isinstance(inp, vtk.vtkAlgorithmOutput):
+        vtk_object.SetInputConnection(inp)
+
+    vtk_object.Update()
+    return vtk_object
+
+
+def map_coordinates_3d_4d(input_array, indices):
```

Comment:
I don't think this should be in the module `viz`. I'm sure this could be useful elsewhere in dipy, not only visualization.